gpm_api/tests/test_io/test_download.py

```diff
@@ -162,74 +165,102 @@ def test_download_file_private(
             )
 
 
-def test_download_data(
-    check,  # For non-failing asserts
-    products: List[str],
-    product_types: List[str],
-    remote_filepaths: Dict[str, Dict[str, Any]],
-    mocker: MockerFixture,
-    versions: List[str],
-):
-    """Test download_data function
-
-    This test is somewhat redundant considering it is testing methods
-    bundled in another functions which need to be turned off in order to
-    test this function. However, it is useful to have a test that checks
-    the entire download process.
-
-    It may be useful as boilerplate to increase the number of tests here in the
-    future.
-    """
-
-    mocker.patch.object(dl, "_download_files", autospec=True, return_value=[])
-    mocker.patch.object(dl, "_download_daily_data", autospec=True, return_value=([], versions))
-    mocker.patch.object(dl, "run", autospec=True, return_value=None)
-    from gpm_api.io import info, pps
-
-    mocker.patch.object(
-        info,
-        "_get_info_from_filename",
-        autospec=True,
-        return_value={
-            "product": "2A-CLIM",
-            "product_type": "CLIM",
-            "start_time": datetime.datetime(2022, 9, 7, 12, 0, 0),
-            "end_time": datetime.datetime(2022, 9, 7, 13, 0, 0),
-            "version": "V07A",
-            "satellite": "GPM",
-            "granule_id": "2A-CLIM.GPM.GMI.GPROF2021v1.20150301-S121433-E134706.005708.V07A.HDF5",
-        },
-    )
-    mocker.patch.object(
-        find,
-        "find_daily_filepaths",
-        autospec=True,
-        return_value=(remote_filepaths.keys(), versions),
-    )
-
-    # Assume files pass file integrity check by mocking return as empty
-    for product_type in product_types:
-        for product in available_products(product_type=product_type):
-            start_time = get_product_start_time(product)
-            if start_time is None:
-                continue
-            res = dl.download_archive(
-                product=product,
-                start_time=start_time,
-                end_time=start_time + datetime.timedelta(hours=1),
-                product_type=product_type,
-            )
+class TestDownloadArchive:
+    @pytest.fixture(autouse=True)
+    def mock_download(
+        self,
+        mocker: MockerFixture,
+        remote_filepaths: Dict[str, Dict[str, Any]],
+        versions: List[str],
+    ) -> None:
+        mocker.patch.object(dl, "_download_files", autospec=True, return_value=[])
+        mocker.patch.object(dl, "_download_daily_data", autospec=True, return_value=([], versions))
+        mocker.patch.object(dl, "run", autospec=True, return_value=None)
+        from gpm_api.io import info, pps
+
+        mocker.patch.object(
+            info,
+            "_get_info_from_filename",
+            autospec=True,
+            return_value={
+                "product": "2A-CLIM",
```

Comment:
I am not entirely sure I grasp correctly what it's going on here. 
For whatever product below is passed in the download function below, `_get_info_from_filename` will return this dummy inexistent product information? Maybe we should mock this function and ensure that the result varies depending on the `product` info in the `product.yaml` file?

Reply:
Indeed, the same information is returned regardless of the filename. It seems the tests in this class were written just to make sure than the whole `download_archive` function is executed without issues, but there are no actual checks.

Reply:
Ok